djreact/djangoapps/trips/admin.py

```diff
@@ -3,14 +3,84 @@
 
 from django.contrib import admin
 
-from models import (
-    Trip, Activity, Facility, Location, TripItinerary, TripSchedule, Host
-)
-
-admin.site.register(Trip)
-admin.site.register(Activity)
-admin.site.register(Facility)
-admin.site.register(Location)
-admin.site.register(TripItinerary)
-admin.site.register(TripSchedule)
-admin.site.register(Host)
+from models import (Activity, Facility, Host, Location, Trip, TripItinerary,
+                    TripSchedule)
+
+
+class TripScheduleAdminInline(admin.TabularInline):
+    """Trip schedule inline modal admin"""
+    model = TripSchedule
+    extra = 0
+
+
+class TripItineraryAdminInline(admin.StackedInline):
+    """Trip itinerary inline modal admin"""
+    model = TripItinerary
+    extra = 0
+
+
+class TripAdmin(admin.ModelAdmin):
+    """Trip modal admin configuration"""
+    inlines = [TripItineraryAdminInline, TripScheduleAdminInline]
+    prepopulated_fields = {'slug': ('name',)}
+    list_display = (
+        'name', 'slug', 'price'
+    )
+    list_filter = ('starting_location',)
+    search_fields = ['name', 'description', 'locations_included']
+
+
+class ActivityAdmin(admin.ModelAdmin):
+    """Activity modal admin configuration"""
+    prepopulated_fields = {'slug': ('name',)}
+    list_display = (
+        'name', 'slug',
+    )
+
+
+class FacilityAdmin(admin.ModelAdmin):
+    """Facility modal admin configuration"""
+    prepopulated_fields = {'slug': ('name',)}
+    list_display = (
+        'name', 'slug',
+    )
+
+
+class LocationAdmin(admin.ModelAdmin):
+    """Location modal admin configuration"""
+    prepopulated_fields = {'slug': ('name',)}
+    list_display = (
+        'name', 'slug',
+    )
+
+
+class TripItineraryAdmin(admin.ModelAdmin):
+    """Trip itinerary modal admin configuration"""
+    list_display = ('trip', 'description')
+    list_filter = ('trip',)
+    search_fields = ['trip']
+
+
+class TripScheduleAdmin(admin.ModelAdmin):
+    """Trip schedule admin configuration"""
+    list_display = ('trip', 'date_from')
+    list_filter = ('date_from',)
+    search_fields = ['trip']
+
+
+class HostAdmin(admin.ModelAdmin):
+    """Host modal admin configuration"""
+    list_display = (
+        'name', 'description', 'verified'
+    )
+    list_filter = ('verified',)
+    search_fields = ['name', 'description']
+
+
+admin.site.register(Trip, TripAdmin)
```

Comment:
I think using decorators on the classes would look nice, rather than creating classes and mentioning the register calls later.

Reply:
`@admin.register(Location)`
`class LocationAdmin(ModelAdmin)`

This is just for readability purpose. The current structure is ok as well.

Reply:
Thank you.